components/CopyAddress.tsx

```diff
@@ -0,0 +1,35 @@
+import copy from "copy-to-clipboard";
+import CopyIcon from "./icons/CopyIcon";
+
+const CopyAddress = ({ address, url }: { address: string; url?: string }) => {
+  return (
+    <div
+      className="-ml-1 inline-flex cursor-pointer items-center px-1 font-mono hover:bg-dark hover:text-white active:bg-darkGray3"
+      onClick={() => {
+        copy(address);
+      }}
+    >
+      <span className="mr-2 hidden lg:block">
+        {url ? (
+          <a href={url} target="_blank" rel="noopener noreferrer">
+            {address}
+          </a>
+        ) : (
+          address
+        )}
+      </span>
+      <span className="mr-2 lg:hidden">
+        {url ? (
+          <a href={url} target="_blank" rel="noopener noreferrer">
+            {address.slice(0, 6) + "..." + address.slice(-6)}
+          </a>
+        ) : (
+          address.slice(0, 6) + "..." + address.slice(-6)
```

Comment:
i'm dumb. can you explain how this span is different than the other one? is it for being responsive in mobile?

Reply:
the `lg:hidden` means that for screen size `lg` and above this span will be hidden and for the other one it's `lg:block` meaning for that it will only appear as a block when screen size is `lg` and above, so at any point only 1 of these spans will be displayed

Reply:
thanks for explaining 🙇